### ISSUE TITLE
Broken output when using mvnd in multimodule project

### ISSUE THREAD

**Issue by asbachb** [apache/netbeans#5479]
### Apache NetBeans version

Apache NetBeans 17 release candidate

### What happened

When execute a maven command with mvnd on a multi module project the output is broken and stops. In addition a `NullPointerException is thrown`.

Output:
```
cd /home/asbachb/dev/extsrc/wildfly; JAVA_HOME=/home/asbachb/.sdkman/candidates/java/17.0.6-tem M2_HOME=/home/asbachb/.sdkman/candidates/mvnd/current /home/asbachb/.sdkman/candidates/mvnd/current/bin/mvnd --no-transfer-progress clean
Processing build on daemon 5e021129
Scanning for projects...
BuildTimeEventSpy is registered.
------------------------------------------------------------------------
Reactor Build Order:

WildFly: Parent Aggregator                                         [pom]
WildFly: Naming Subsystem                                          [jar]
WildFly: EE 9 Source Transform parent                              [pom]
WildFly: EE subsystem                                              [jar]
WildFly: Application Client Bootstrap                              [jar]
WildFly: Weld Subsystem SPI                                        [jar]
WildFly: Weld Common Tools                                         [jar]
WildFly: IIOP Openjdk Subsystem                                    [jar]
WildFly: Transaction Subsystem                                     [jar]
WildFly: Batch Integration Subsystem (JBeret implementation)       [jar]
WildFly: Common Dependency Management (Base Dependencies)          [pom]
WildFly: Common Dependency Management (Expansion Dependencies)     [pom]
WildFly: Legacy Dependency Management (Base Dependencies)          [pom]
WildFly: Legacy Dependency Management (Expansion Dependencies)     [pom]
WildFly Preview: Dependency Management (Base Dependencies)         [pom]
WildFly Preview: Dependency Management (Expansion Dependencies)    [pom]
WildFly: Dependency Management (Base Dependencies)                 [pom]
WildFly: Dependency Management (Expansion Dependencies)            [pom]
WildFly: Web Common Classes                                        [jar]
WildFly: Elytron OpenID Connect Client Parent                      [pom]
WildFly: Elytron OpenID Connect Client Extension                   [jar]
WildFly: Galleon Pack OpenID Connect Common                        [pom]
WildFly: Clustering subsystems and modules                         [pom]
WildFly: Contextual execution for clustering modules               [jar]
WildFly: Clustering services                                       [jar]
WildFly: Common code for clustering subsystems                     [jar]
WildFly: Web session clustering                                    [pom]
WildFly: Web session clustering - Container SPI                    [jar]
WildFly: Security Subsystem parent                                 [pom]
WildFly: Security Subsystem                                        [jar]
WildFly: Undertow                                                  [jar]
WildFly: Servlet Feature Pack Parent                               [pom]
WildFly: Servlet Feature Pack Common Resources                     [pom]
wildfly-servlet-feature-pack-galleon-common                        [pom]
wildfly-servlet-feature-pack-galleon-legacy                        [pom]
WildFly: Jipijapa SPI                                              [jar]
WildFly: Jipijapa Eclipselink                                      [jar]
WildFly: JPA                                                       [pom]
WildFly: Jipijapa Hibernate 6 (JPA 3.1) integration                [jar]
WildFly: Bean Validation subsystem                                 [jar]
WildFly: Infinispan modules                                        [pom]
WildFly: Infinispan embedded modules                               [pom]
WildFly: Infinispan embedded API extensions                        [jar]
WildFly: Infinispan embedded services                              [jar]
WildFly: EE clustering                                             [pom]
WildFly: EE clustering SPI                                         [jar]
WildFly: Clustering marshalling modules                            [pom]
WildFly: Clustering marshalling API                                [jar]
WildFly: Clustering marshalling SPI                                [jar]
WildFly: Marshalling for WildFly clustering: ProtoStream integration [jar]
WildFly: Common EE implementations for caches                      [jar]
WildFly: Marshalling for WildFly clustering: JBoss Marshalling integration [jar]
WildFly: Server clustering modules                                 [pom]
WildFly: Public server clustering API                              [jar]
WildFly: Server clustering SPI                                     [jar]
WildFly: Server clustering requirements/services                   [jar]
WildFly: Connector Subsystem                                       [jar]
WildFly: SFSB clustering                                           [pom]
WildFly: SFSB clustering - SPI                                     [jar]
WildFly: Infinispan marshalling                                    [jar]
WildFly: Infinispan embedded SPI                                   [jar]
WildFly: Singleton modules                                         [pom]
WildFly: Singleton API                                             [jar]
WildFly: EJB Subsystem                                             [jar]
WildFly: JPA Subsystem                                             [jar]
WildFly: Jakarta Enterprise Beans and Jakarta Messaging client combined properties [jar]
WildFly: Jakarta Enterprise Beans and Jakarta Messaging client combined jar [jar]
WildFly: Clustering Jakarta Expression Language API modules        [pom]
WildFly: Clustering support for the Eclipse Expressly implementation of the Jakarta Expression Language API [jar]
WildFly: Clustering Jakarta Faces modules                          [pom]
WildFly: Clustering support for the Jakarta Faces API              [jar]
WildFly: Clustering support for the Mojarra implementation of Jakarta Faces [jar]
WildFly: Clustering Weld modules                                   [pom]
WildFly: Clustering support for Weld                               [jar]
WildFly: EJB client clustering module                              [jar]
WildFly: Weld                                                      [pom]
WildFly: Weld EJB                                                  [jar]
WildFly: Clustering support for Weld's ejb module                  [jar]
WildFly: Clustering support for Weld's web module                  [jar]
WildFly: Web session clustering API                                [jar]
WildFly: Web session clustering SPI                                [jar]
WildFly: Web session clustering requirements and service providers [jar]
WildFly: Web session clustering - Undertow integration             [jar]
WildFly: Datasources with Agroal connection pool                   [jar]
WildFly: Jakarta EE Security                                       [jar]
WildFly: JacORB Subsystem                                          [jar]
WildFly: JAX-RS Integration                                        [jar]
WildFly: Jakarta Faces integration modules                         [pom]
WildFly: JSF Subsystem                                             [jar]
WildFly: JSF Injection Handlers                                    [jar]
WildFly: Keycloak Subsystem                                        [jar]
WildFly: Web Subsystem                                             [jar]
WildFly: JSR-77 Subsystem                                          [jar]
WildFly: Mail subsystem                                            [jar]
WildFly: JGroups modules                                           [pom]
WildFly: JGroups API                                               [jar]
WildFly: JGroups SPI                                               [jar]
WildFly: Messaging Injection With ActiveMQ Artemis                 [jar]
WildFly: JGroups Subsystem                                         [jar]
WildFly: Messaging Subsystem With ActiveMQ Artemis                 [jar]
WildFly: (Legacy) Messaging Subsystem                              [jar]
WildFly: mod_cluster Subsystem                                     [pom]
WildFly: mod_cluster Extension                                     [jar]
WildFly: mod_cluster Undertow Integration                          [jar]
WildFly: PicketLink Subsystem                                      [jar]
WildFly: OpenTelemetry API                                         [jar]
WildFly: OpenTelemetry                                             [jar]
WildFly: Web Services Subsystem                                    [pom]
WildFly: Web Services Server Integration Subsystem                 [jar]
WildFly: Web Services OpenSAML ConfigurationPropertiesSource Impl  [jar]
WildFly: Weld Bean Validation                                      [jar]
WildFly: Weld Subsystem                                            [jar]
WildFly: Weld JPA                                                  [jar]
WildFly: Weld Transactions                                         [jar]
WildFly: Weld Webservices                                          [jar]
WildFly: RTS Subsystem                                             [jar]
WildFly: XTS Subsystem                                             [jar]
WildFly: System JMX Module                                         [jar]
WildFly: EE clustering - HotRod service provider                   [jar]
WildFly: EE clustering - Infinispan service provider               [jar]
WildFly: SFSB clustering - Infinispan integration                  [jar]
WildFly: Distributable EJB Subsystem                               [jar]
WildFly: Infinispan client modules                                 [pom]
WildFly: Infinispan Client API                                     [jar]
WildFly: Infinispan Client requirements and services               [jar]
WildFly: Infinispan Client SPI                                     [jar]
WildFly: Infinispan subsystem                                      [jar]
WildFly: Server clustering SPI implementation                      [jar]
WildFly: Server clustering extension                               [jar]
WildFly: Requirements for WildFly clustering singleton service configurators [jar]
WildFly: Singleton extension                                       [jar]
WildFly: WildFly clustering singleton service implementation       [jar]
WildFly: Common abstractions for cache-based session manager implementations. [jar]
WildFly: Web session clustering - HotRod service provider          [jar]
WildFly: Web session clustering - Infinispan service provider      [jar]
WildFly: Distributable Web Subsystem                               [jar]
WildFly: Base Health Extension                                     [jar]
WildFly: JDR                                                       [jar]
WildFly: Base Metrics Extension                                    [jar]
WildFly: POJO Subsystem                                            [jar]
WildFly: Service Archive Subsystem                                 [jar]
WildFly: EE Feature Pack Parent                                    [pom]
WildFly: EE Feature Pack Common Resources                          [pom]
WildFly: Preview Feature Pack EE 10 API Resources                  [pom]
WildFly: EE Feature Pack Galleon Common                            [pom]
WildFly: EE Feature Pack Galleon Content                           [pom]
WildFly: EE Full Galleon Pack                                      [pom]
WildFly: MicroProfile modules                                      [pom]
WildFly: MicroProfile Fault Tolerance modules                      [pom]
WildFly: MicroProfile Fault Tolerance - Executor                   [jar]
WildFly: MicroProfile Fault Tolerance - Extension                  [jar]
WildFly: Microprofile integration modules                          [pom]
WildFly: MicroProfile JWT Extension With SmallRye                  [jar]
WildFly: MicroProfile OpenAPI Extension With SmallRye              [jar]
WildFly: MicroProfile Config Extension With SmallRye               [jar]
WildFly: MicroProfile OpenTracing Extension With SmallRye          [jar]
WildFly: MicroProfile OpenTracing Extension With SmallRye          [jar]
WildFly: MicroProfile Reactive Streams Operators CDI Provider      [jar]
WildFly: MicroProfile Health Extension With SmallRye               [jar]
WildFly: MicroProfile Metrics Extension With SmallRye              [jar]
WildFly: MicroProfile Reactive Messaging Parent                    [pom]
WildFly: MicroProfile Reactive Messaging Common                    [jar]
WildFly: MicroProfile Reactive Messaging Config                    [jar]
WildFly: MicroProfile Reactive Messaging Kakfa                     [jar]
WildFly: MicroProfile Reactive Messaging Extension With SmallRye   [jar]
WildFly: MicroProfile Reactive Streams Operators Parent            [pom]
WildFly: MicroProfile Reactive Streams Operators Extension With SmallRye [jar]
WildFly: Galleon Pack MicroProfile Common                          [pom]
WildFly: Feature Pack Parent                                       [pom]
WildFly: Feature Pack Galleon Content                              [pom]
WildFly: Galleon Pack                                              [pom]
WildFly: Thin Server Build                                         [pom]
WildFly Test Suite: Shared                                         [jar]
WildFly: Distribution                                              [jar]
WildFly: EE Full Thin Server Build                                 [pom]
WildFly: EE Full Distribution                                      [jar]
WildFly: EE Feature Pack Pruned Resources                          [pom]
WildFly Preview Test Suite: Shared                                 [jar]
WildFly Preview: Parent                                            [pom]
WildFly: EE 9 Deployment Transformer                               [jar]
WildFly: Observability modules                                     [pom]
WildFly: Micrometer API                                            [jar]
WildFly: EE                                                        [jar]
WildFly: Web Common Classes                                        [jar]
WildFly: Base Micrometer Extension                                 [jar]
WildFly: EE Preview Feature Pack Common Resources                  [pom]
WildFly Preview: Galleon Feature Pack Common Microprofile Resources [pom]
WildFly Preview: Galleon Feature Pack                              [pom]
WildFly: EE 9 Preview Thin Server Build                            [pom]
WildFly: EE 9 Preview Distribution                                 [jar]
WildFly: Preview Feature Pack Content Resources                    [pom]
WildFly: Preview Feature Pack Content MicroProfile Resources       [pom]
WildFly: Weld Subsystem SPI                                        [jar]
WildFly: Weld Common Tools                                         [jar]
WildFly: Bean Validation                                           [jar]
WildFly: Jipijapa SPI                                              [jar]
WildFly: JPA Subsystem                                             [jar]
WildFly: Jipijapa EclipseLink integration                          [jar]
WildFly: MicroProfile Reactive Streams Operators CDI Provider      [jar]
WildFly: Undertow                                                  [jar]
WildFly: MicroProfile Health Extension With SmallRye               [jar]
WildFly: MicroProfile JWT Extension With SmallRye                  [jar]
WildFly: Weld Subsystem                                            [jar]
WildFly: MicroProfile OpenTracing with SmallRye                    [jar]
WildFly: MicroProfile OpenTracing Extension                        [jar]
WildFly: Release                                                   [pom]
WildFly: Exported Jakarta EE Specification APIs                    [pom]
WildFly: Validation Tests for Exported Jakarta EE Specification APIs [jar]
WildFly: Weld JPA                                                  [jar]
WildFly: Weld Bean Validation                                      [jar]
WildFly: Weld Transactions                                         [jar]
WildFly: Test Feature Pack                                         [pom]
WildFly Test Suite: Aggregator                                     [pom]
WildFly Test Suite: Integration (parent)                           [pom]
WildFly Test Suite: Integration - Web                              [jar]
WildFly Test Suite: Integration - Smoke                            [jar]
WildFly: IIOP Openjdk Subsystem                                    [jar]
WildFly: Transaction Subsystem                                     [jar]
WildFly: Connector Subsystem                                       [jar]
WildFly: JSF                                                       [pom]
WildFly: JSF Subsystem                                             [jar]
WildFly: JSF Injection Handlers                                    [jar]
WildFly: JSF Multi-JSF installer                                   [jar]
WildFly: Messaging Subsystem Parent With ActiveMQ Artemis          [pom]
WildFly: Messaging Injection With ActiveMQ Artemis                 [jar]
WildFly: Messaging Subsystem With ActiveMQ Artemis                 [jar]
WildFly: OpenTelemetry API                                         [jar]
WildFly: Base OpenTelemetry Extension                              [jar]
WildFly: RTS Subsystem                                             [jar]
WildFly: Web Services Tests Integration Subsystem                  [jar]

Using the SmartBuilder implementation with a thread count of 3
Task segments : [clean]
Build maximum degree of concurrency is 3
Total number of projects is 230
[wildfly-parent] 
[wildfly-parent] [INFO] 
[wildfly-parent] [INFO] ---------------------< org.wildfly:wildfly-parent >---------------------
[wildfly-parent] [INFO] Building WildFly: Parent Aggregator 27.0.1.Final                 [1/230]
[wildfly-parent] [INFO] --------------------------------[ pom ]---------------------------------
[wildfly-parent] 
[wildfly-parent] [INFO] 
[wildfly-parent] [INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ wildfly-parent ---
[wildfly-parent] [INFO] Deleting /home/asbachb/dev/extsrc/wildfly/target
[wildfly-parent] 
[wildfly-parent] 
[wildfly-clustering] 
[wildfly-clustering] [INFO] 
[wildfly-clustering] [INFO] -------------------< org.wildfly:wildfly-clustering >-------------------
[wildfly-clustering] [INFO] Building WildFly: Clustering subsystems and modules 27.0.1.Final [2/230]
[wildfly-clustering] [INFO] --------------------------------[ pom ]---------------------------------
[wildfly-ee-9-source-transform-parent] 
[wildfly-ee-9-source-transform-parent] [INFO] 
[wildfly-ee-9-source-transform-parent] [INFO] ----------< org.wildfly:wildfly-ee-9-source-transform-parent >----------
[wildfly-ee-9-source-transform-parent] [INFO] Building WildFly: EE 9 Source Transform parent 27.0.1.Final      [3/230]
[wildfly-ee-9-source-transform-parent] [INFO] --------------------------------[ pom ]---------------------------------
[wildfly-naming] 
[wildfly-naming] [INFO] 
[wildfly-naming] [INFO] ---------------------< org.wildfly:wildfly-naming >---------------------
[wildfly-ee-9-source-transform-parent] 
[wildfly-ee-9-source-transform-parent] [INFO] 
[wildfly-ee-9-source-transform-parent] [INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ wildfly-ee-9-source-transform-parent ---
[wildfly-ee-9-source-transform-parent] 
[wildfly-naming] [INFO] Building WildFly: Naming Subsystem 27.0.1.Final                  [4/230]
[wildfly-naming] [INFO] --------------------------------[ jar ]---------------------------------
[wildfly-ee-9-source-transform-parent] 
[wildfly-microprofile-legacy-namespace] 
[wildfly-microprofile-legacy-namespace] [INFO] 
[wildfly-microprofile-legacy-namespace] [INFO] ---------< org.wildfly:wildfly-microprofile-legacy-namespace >----------
[wildfly-microprofile-legacy-namespace] [INFO] Building WildFly: MicroProfile modules 27.0.1.Final              [5/230]
[wildfly-microprofile-legacy-namespace] [INFO] --------------------------------[ pom ]---------------------------------
[wildfly-naming] 
[wildfly-naming] [INFO] 
[wildfly-naming] [INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ wildfly-naming ---
[wildfly-microprofile-legacy-namespace] 
[wildfly-microprofile-legacy-namespace] [INFO] 
[wildfly-microprofile-legacy-namespace] [INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ wildfly-microprofile-legacy-namespace ---
[wildfly-naming] 
[wildfly-clustering] 
[wildfly-clustering] [INFO] 
[wildfly-clustering] [INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ wildfly-clustering ---
[wildfly-naming] 
[wildfly-weld-parent] 
[wildfly-weld-parent] [INFO] 
[wildfly-weld-parent] [INFO] ------------------< org.wildfly:wildfly-weld-parent >-------------------
[wildfly-weld-parent] [INFO] Building WildFly: Weld 27.0.1.Final                              [6/230]
[wildfly-weld-parent] [INFO] --------------------------------[ pom ]---------------------------------
[wildfly-weld-parent] 
[wildfly-weld-parent] [INFO] 
[wildfly-clustering] 
[wildfly-weld-parent] [INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ wildfly-weld-parent ---
[wildfly-clustering] 
[wildfly-iiop-openjdk] 
[wildfly-iiop-openjdk] [INFO] 
[wildfly-iiop-openjdk] [INFO] ------------------< org.wildfly:wildfly-iiop-openjdk >------------------
[wildfly-iiop-openjdk] [INFO] Building WildFly: IIOP Openjdk Subsystem 27.0.1.Final            [7/230]
[wildfly-iiop-openjdk] [INFO] --------------------------------[ jar ]---------------------------------
[wildfly-microprofile-legacy-namespace] 
``` 

Stacktrace:
```
java.lang.NullPointerException: Cannot read field "type" because "start" is null
	at org.netbeans.modules.maven.execute.CommandLineOutputHandler.trimTree(CommandLineOutputHandler.java:575)
	at org.netbeans.modules.maven.execute.CommandLineOutputHandler.processExecEvent(CommandLineOutputHandler.java:480)
	at org.netbeans.modules.maven.execute.CommandLineOutputHandler.access$400(CommandLineOutputHandler.java:75)
	at org.netbeans.modules.maven.execute.CommandLineOutputHandler$Output.run(CommandLineOutputHandler.java:286)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1418)
	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:45)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:287)
	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2033)
Caused: org.openide.util.RequestProcessor$SlowItem
	at org.openide.util.RequestProcessor.post(RequestProcessor.java:395)
	at org.netbeans.modules.maven.execute.CommandLineOutputHandler.setStdOut(CommandLineOutputHandler.java:158)
	at org.netbeans.modules.maven.execute.MavenCommandLineExecutor.executeProcess(MavenCommandLineExecutor.java:416)
	at org.netbeans.modules.maven.execute.MavenCommandLineExecutor.run(MavenCommandLineExecutor.java:354)
	at org.netbeans.core.execution.RunClassThread.doRun(RunClassThread.java:132)
	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:45)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:287)
[catch] at org.netbeans.core.execution.RunClassThread.run(RunClassThread.java:81)

```

### How to reproduce

* git clone https://github.com/wildfly/wildfly.git
* open parent project in NetBeans
* * Configure mvnd 0.9.0 in NetBeans settings
* Click clean on the project

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Ubuntu 22.04 LTS

### JDK

17

### Apache NetBeans packaging

Apache NetBeans provided installer

### Are you willing to submit a pull request?

Yes

### Code of Conduct

Yes

**Comment by akronenw** [comment]
Hello @asbachb, 
when I build my multi pom project, it also stops with the following exception:
`
java.lang.NullPointerException: Cannot invoke "org.openide.windows.FoldHandle.silentStartFold(boolean)" because "parentProject.foldHandle" is null
	at org.netbeans.modules.maven.execute.cmd.ExecutionEventObject$Tree.startFold(ExecutionEventObject.java:179)
	at org.netbeans.modules.maven.execute.CommandLineOutputHandler$Output.run(CommandLineOutputHandler.java:353)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1418)
	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:45)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:278)
[catch] at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2033)
`
Do you think that is also fixed by your PR?
Best
Alex

**Comment by asbachb** [comment]
@akronenw base on your stack trace I don't think so. I could give it a try if you prepare a test project. Alternatively you could build my branch and test it your self. You just need `java 11`+ and `apache ant`.

**Comment by mbien** [comment]
@akronenw should be fixed by #4957
